Assets/Editor/CreatureHydrator.cs

```diff
@@ -224,7 +224,13 @@ private static void HydrateCreature(UnitDefinition unit, string generaName, stri
         
         // Load the newly generated script
         MonoScript newScript = AssetDatabase.LoadAssetAtPath<MonoScript>(scriptPath);
-        
+        AssetDatabase.Refresh();
+        // Need to compile the new script file before we can attach it to the prefab
+        while (EditorApplication.isCompiling)
+        {
+            Debug.Log("Waiting for Unity to finish compiling...");
+            System.Threading.Thread.Sleep(100);  // Wait for a short time before checking again
```

Comment:
noooooooooooooOOOOOOoooooooooooooo

Reply:
Ah wait I see that boolean, nevermind 😆

Reply:
i removed the sleep in favor doing hydrating and linking separately